src/OrbitGl/TimelineUi.cpp

```diff
@@ -59,15 +59,22 @@ void TimelineUi::RenderLabels(Batcher& batcher, TextRenderer& text_renderer,
     }
     float world_x = timeline_info_interface_->GetWorldFromUs(
         tick_ns / static_cast<double>(kNanosecondsPerMicrosecond));
-    if (world_x > previous_label_end_x) {
+    if (world_x > previous_label_end_x + kLabelMarginRight + kPixelMargin) {
       Vec2 pos, size;
-      float label_bottom_y = GetPos()[1] + GetHeight() - kLabelMarginBottom;
+      float label_bottom_y =
```

Comment:
You can use `TextRenderer::VAlign::Middle` in the following `AddText` call, then you don't need this calculation.

Reply:
Yes, true! :)

Reply:
Oh, I just realize this doesn't make it a lot easier because you still need to calculate the middle line... I thought "Middle" allowed you to specify a top and height, and it would automatically center the text in the middle. Anyways, this is fine now.